File: Day 9/Majority Element.py
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> mpp= new HashMap<>();
        int res=0; //to store the char
        int majority = 0; //to store the freq

        for (int n:nums){
            mpp.put(n, 1+ mpp.getOrDefault(n,0));
            /* int a = mpp.get(n);
            a = Math.max(a,majority);
            res = n; */
            if (mpp.get(n) > majority){
                 res = n;
                 majority = mpp.get(n); 
            }
        }
        return res;
    }
}

File: PLACEMENT-PREP/Striver's SDE Sheet/Majority-Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        HashMap<Integer,Integer> mpp = new HashMap<>();

        for (int num : nums){
            mpp.put(num, mpp.getOrDefault(num,0) +1);
        }

        for (Map.Entry<Integer, Integer> entry : mpp.entrySet()) {
            if (entry.getValue() > n / 2) {
                return entry.getKey();
            }
        }
        return -1;
    }
}
